romeo/src/stacks_client.rs

```diff
@@ -0,0 +1,247 @@
+//! Stacks client
+
+use anyhow::anyhow;
+use bdk::bitcoin::Network;
+use blockstack_lib::codec::StacksMessageCodec;
+use blockstack_lib::core::CHAIN_ID_TESTNET;
+use rand::distributions::Alphanumeric;
+use rand::{thread_rng, Rng};
+use reqwest::{Request, StatusCode};
+use serde_json::Value;
+use std::sync::Arc;
+use tokio::sync::{Mutex, MutexGuard};
+use tracing::trace;
+
+use serde::de::DeserializeOwned;
+
+use blockstack_lib::burnchains::Txid as StacksTxId;
+use blockstack_lib::chainstate::stacks::{
+    StacksTransaction, StacksTransactionSigner, TransactionAnchorMode, TransactionPostConditionMode,
+};
+use blockstack_lib::{
+    address::{
+        AddressHashMode, C32_ADDRESS_VERSION_MAINNET_SINGLESIG,
+        C32_ADDRESS_VERSION_TESTNET_SINGLESIG,
+    },
+    types::chainstate::{StacksAddress, StacksPrivateKey, StacksPublicKey},
+};
+
+/// Wrapped Stacks Client which can be shared safely between threads.
+#[derive(Clone, Debug)]
+pub struct LockedClient(Arc<Mutex<StacksClient>>);
```

Comment:
Is this something that should maybe go in the SDK? I wrote the same thing as part of block producer code (minus the wrapping for mulithreading). Though the signing portion would have to be a trait that could be overwritten to accommodate FROST for example.

Reply:
Yeah this belongs in the SDK but to save a bit of time we've just put it here for now.

Reply:
https://github.com/stacks-network/sbtc/issues/80